packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -183,6 +178,19 @@ function computeEntryTracingInfo(entry: RumPerformanceResourceTiming, configurat
   }
 }
 
+function computeResponseDurationInfo(request: RequestCompleteEvent) {
+  let durationDiff
+  let durationPercentageDiff
+  if (request.resolveDuration) {
+    durationDiff = toServerDuration((request.duration - request.resolveDuration) as Duration)
+    durationPercentageDiff = Math.round((durationDiff / toServerDuration(request.duration)) * 100)
+  }
+  return {
+    durationDiff,
+    durationPercentageDiff,
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about encapsulating all the response duration info logic/format in this method and return directly:
```ts
return {
  _dd: {
    resolveDuration,
    durationDiff,
    durationPercentageDiff
  }
}
```
it is what we are doing on other methods